### PR TITLE
feat: Implement awala member key import request

### LINKED ISSUE #97: Process member public key imports in Awala endpoint
`POST /awala` with content type `application/vnd.veraid.member-public-key-import`. As described on the README.

- [x] Call `createMemberPublicKey()`.
- [x] Publish event (following example in `exampleEventPublisher.routes.ts`). Parameters:
  - id: MongoDB id of the member. For efficiency reasons, in case we get multiple imports from the same member in quick succession -- that way, the broker will take the latest.
  - type: `net.veraid.authority.member-public-key-import`
  - source: `https://veraid.net/authority/awala-member-key-import`
  - data: The id of the public key and the Awala PDA.
- [x] Remove `exampleEventPublisher.routes.ts`.

### PR DESCRIPTION
**Pull request by levansuper**
closes #97

### CODE REVIEW

File: src/api/routes/awala.routes.ts [hunk 1]
```diff
@@ -50,14 +54,48 @@ async function processMemberBundleRequest(
...
+): Promise<boolean> {
+  const validationResult = validateMessage(data, MEMBER_KEY_IMPORT_REQUEST_SCHEMA);
+  if (typeof validationResult === 'string') {
+    options.logger.info(
+      {
+        publicKeyImportToken: (
+          data as {
+            publicKeyImportToken: string;
+          }
+        ).publicKeyImportToken,
```
[the remark this answers is not in the document]
Comment (levansuper) [reply src/api/routes/awala.routes.ts:69]: as any is restricted by the linter that's why I implemented it this way

File: src/memberKeyImportToken.ts [hunk 2]
```diff
@@ -28,3 +37,60 @@ export async function createMemberKeyImportToken(
...
+    },
+    options,
+  );
+
+  if (!craetePublicKeyResult.didSucceed) {
+    return {
+      didSucceed: false,
+      reason: MemberPublicKeyImportProblemType.KEY_CREATION_ERROR,
+    };
+  }
```
[the remark this answers is not in the document]
Comment (levansuper) [reply src/memberKeyImportToken.ts:73]: This happens if the public key is invalid

File: src/memberKeyImportToken.ts [hunk 3]
```diff
@@ -28,3 +37,60 @@ export async function createMemberKeyImportToken(
...
+      publicKeyId: craetePublicKeyResult.result.id,
+      awalaPda: keyImportData.awalaPda,
+    },
+  });
+  await emitter.emit(event);
+
+  await memberKeyImportTokenModel.findByIdAndDelete(keyImportData.publicKeyImportToken);
+  options.logger.info(
+    { id: keyImportData.publicKeyImportToken },
+    'Member public key import token deleted',
```
[the remark this answers is not in the document]
Comment (levansuper) [reply src/memberKeyImportToken.ts:91]: This part is logged in the create public key method. Should we log here too?

File: src/api/routes/awala.routes.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (gnarea) [reply src/api/routes/awala.routes.ts:69]: I believe that's limited to using `any` values inadvertently. Explicitly marking it `as any` should be fine, for the same reason that explicitly marking it `as {foo: bar}` fine: `unknown` can be "cast" to any type.

File: src/memberKeyImportToken.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (gnarea) [reply src/memberKeyImportToken.ts:73]: You're right, I must've been looking at a different function by mistake 😬

File: src/memberKeyImportToken.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (gnarea) [reply src/memberKeyImportToken.ts:91]: Cool, in that case the current log message makes sense 👌🏾

File: src/api/routes/awala.routes.spec.ts [hunk 4]
```diff
@@ -132,10 +185,53 @@ describe('awala routes', () => {
...
+      });
+
+      const response = await serverInstance.inject({
+        method: 'POST',
+        url: '/awala',
+        headers: validHeaders,
+        payload: validPayload,
+      });
+
+      expect(response).toHaveProperty('statusCode', HTTP_STATUS_CODES.BAD_REQUEST);
```
[the remark this answers is not in the document]
Comment (levansuper) [reply src/api/routes/awala.routes.spec.ts:230]: We are not returning anything in the response. should we?

[the remark this answers is not in the document]
Comment (gnarea) [reply src/api/routes/awala.routes.spec.ts:230]: Cool, let's ignore my comment then

File: src/memberKeyImportToken.ts [hunk 5]
```diff
@@ -50,13 +50,17 @@ export async function processMemberKeyImportToken(
     keyImportData.publicKeyImportToken,
   );
   if (!memberKeyImportToken) {
+    options.logger.info(
+      { id: keyImportData.publicKeyImportToken },
```
Comment (gnarea) [inline src/memberKeyImportToken.ts:54]: We refer to this as "token", not "id", so I think we should be consistent. "Id" is too ambiguous -- is the the member id?

**Review by gnarea — CHANGES_REQUESTED** [review]

**Review by gnarea — APPROVED** [review]
Thanks!

**Pull request merged**